presto-main/src/main/java/com/facebook/presto/execution/scheduler/group/DynamicBucketNodeMap.java

```diff
@@ -75,4 +80,18 @@ public boolean isDynamic()
     {
         return true;
     }
+
+    @Override
+    public Optional<List<InternalNode>> getBucketToNode()
+    {
+        if (bucketToNode.size() == 0) {
+            return Optional.empty();
+        }
+        return Optional.of(ImmutableList.copyOf(bucketToNode.values()));
+    }
+
+    public boolean ifInitialMapExist()
```

Comment:
I think this could be an interface method, and for `FixedBucketNodeMap` just return `true`. This way we don't need to do downcasting :)

Also, maybe name this `hasInitialMap`?